lib/mini_twitter_client.rb

```diff
@@ -0,0 +1,81 @@
+require 'faraday'
+require 'json'
+
+class Tweet
+  def initialize(name = 'name', email = 'email@email.com', message = 'message', id = nil)
+    @name = name
+    @email = email
+    @message = message
+    @id =id
+  end
+  def name
+    @name
+  end
+  def email
+    @email
+  end
+  def message
+    @message
+  end
+  def id
+    @id
+  end
+end
+
+class MiniTwitterClient
+  def initialize
+    @connection = Faraday.new(
+      url: 'http://localhost:4000',
+      params: {},
+      headers: {
+        'Accept' => 'application/json',
+        'Content-Type' => 'application/json'
+      }
+    )
+  end
+
+  def get_tweets
+    res = @connection.get('/api/tweets')
+    data = JSON.parse(res.body)['data'].map do |tweet|
+      Tweet.new(tweet['author']['name'], tweet['author']['email'], tweet['message'], tweet['id'])
+    end
+
+    {
+      status: res.status.to_i,
+      data:
+    }
+  end
+
+  def create_tweet(name, email, message)
+    res = @connection.post('/api/tweets') do |post|
+      post.body = { "tweet": { "author": { "name": name, "email": email }, "message": message } }.to_json
+    end
+    tweet = JSON.parse(res.body)['data']
+
+    {
+      status: res.status.to_i,
+      data: Tweet.new(tweet['author']['name'], tweet['author']['email'], tweet['message'], tweet['id'])
```

Comment:
This code is duplicated in more places and could be moved to a single method.

One could say that this method belongs to `Tweet` class. So, in this case, you could use the following pattern:

```suggestion
      data: Tweet.build_from_hash(tweet)
```

and add the class method (https://dev.to/adamlombard/ruby-class-methods-vs-instance-methods-4aje) to `Tweet`:

```ruby
class Tweet
  def self.build_from_hash(tweet)
    Tweet.new(
      id: tweet[:id],     # I assumed you would use keywords arguments in Tweet#initialize definition
      # ..
    )
  end

  # ...
end
```

Reply:
Very good idea.

Reply:
One question, if I'm agree with your suggestion, should I click "Resolve conversation" ? I'm doing this, but I'm not sure is this is ok, maybe You can first read my answer?

Reply:
You can resolve. Personally I like to resolve comments after applying the changes (so I won't forget to do it), but there are no strict rules

Reply:
Ok, I,m doing that as well, but I'm just apply changes in vs code because I like to write everything myself instead of just click commit, at least for now :)